raiden-ts/src/utils/types.ts

```diff
@@ -13,18 +13,37 @@ export interface Storage {
   removeItem(key: string): void | Promise<void>;
 }
 
+/**
+ * Error for assertion functions/type guards
+ */
+export class AssertionError extends Error {}
+
+/**
+ * Type-safe assertion function (TS3.7)
+ *
+ * @param condition - Condition to validate as truthy
+ * @param msg - Message to throw if condition is falsy
+ */
+export function assert(condition: any, msg?: string): asserts condition {
```

Comment:
At some point we should use this everywhere and enforce error codes, that the dApp and other implementers can expect and react on 👍

Reply:
Yes, I agree. We're coding for the minimum amount of errors to occur, and when they do, it's usually either an indicator or a wrong call (e.g. malformed address) or something trully unexpected, when most of what can be done is show it to the user and continue. But surely we could be more specific if any dev using the SDK wants to code specifically for some of these issues.

Reply:
We actually have an issue for the dApp for this, and having documented error codes would be very beneficial because if we return messages we assume that we know how to explain to whatever end-user what went wrong.

With an error code the end-developer has much more control on how to react (e.g. write their own error messages) on errors, that's what I meant